### PR TITLE
switch to statementSemantic()

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by RazvanN7** [comment]
Are we doing this for all semantic AST node classes? Otherwise, what is the point in doing this just for statement nodes?

**Comment by WalterBright** [comment]
All of them.

**Review by JinShil — APPROVED** [review]

File: src/ddmd/semantic.d [hunk 1]
```diff
@@ -54,7 +54,7 @@ extern(C++) Initializer semantic(Initializer init, Scope* sc, Type t, NeedInterp
 }
 
 // Performs semantic analysis in Statement AST nodes
-extern(C++) Statement semantic(Statement s, Scope* sc)
+extern(C++) Statement xxsemantic(Statement s, Scope* sc)
```
Comment (ZombineDev) [inline src/ddmd/semantic.d]: I guess now that you've found and replaced all of the call-sites this function can now be removed.

**Review by PetarKirov — APPROVED** [review]

File: src/ddmd/semantic.d (hunk 1 above)
Comment (WalterBright) [reply src/ddmd/semantic.d]: If I remove it now, then the other PRs will have merge conflicts, and I gotta do a rebase. Simpler to just do it later.

**Pull request merged**